hoodie-common/src/main/java/com/uber/hoodie/common/util/collection/converter/DefaultValueConverter.java

```diff
@@ -0,0 +1,135 @@
+  // The client implementation of HoodieRecordPayload used to re-create HoodieRecord from bytes
+  private final String payloadClazz;
+
+  private static Logger log = LogManager.getLogger(DefaultValueConverter.class);
+
+  public DefaultValueConverter(Schema schema, String payloadClazz) {
+    this.schema = schema;
+    this.payloadClazz = payloadClazz;
+  }
+
+  @Override
+  public byte[] getBytes(HoodieRecord hoodieRecord) {
+    try {
+      // Need to initialize this to 0 bytes since deletes are handled by putting an empty record in HoodieRecord
+      byte[] val = new byte[0];
+      if (hoodieRecord.getData().getInsertValue(schema).isPresent()) {
+        val = HoodieAvroUtils
+            .avroToBytes((GenericRecord) hoodieRecord.getData().getInsertValue(schema).get());
+      }
+      Pair<Pair<String, String>, byte[]> data =
+          Pair.of(Pair.of(hoodieRecord.getKey().getRecordKey(),
+              hoodieRecord.getKey().getPartitionPath()), val);
+      return SerializationUtils.serialize(data);
+    } catch (IOException io) {
+      throw new HoodieNotSerializableException("Cannot serialize value to bytes", io);
+    }
+  }
+
+  @Override
+  public HoodieRecord getData(byte[] bytes) {
+    try {
+      Pair<Pair<String, String>, byte[]> data = SerializationUtils.deserialize(bytes);
+      Optional<GenericRecord> payload = Optional.empty();
+      if (data.getValue().length > 0) {
+        // This can happen if the record is deleted, the payload is optional with 0 bytes
+        payload = Optional.of(HoodieAvroUtils.bytesToAvro(data.getValue(), schema));
+      }
+      HoodieRecord<? extends HoodieRecordPayload> hoodieRecord = new HoodieRecord<>(
+          new HoodieKey(data.getKey().getKey(), data.getKey().getValue()),
+          ReflectionUtils
+              .loadPayload(payloadClazz,
+                  new Object[]{payload}, Optional.class));
+      return hoodieRecord;
+    } catch (IOException io) {
+      throw new HoodieNotSerializableException("Cannot de-serialize value from bytes", io);
+    }
+  }
+
+  @Override
+  public long sizeEstimate(HoodieRecord<? extends HoodieRecordPayload> hoodieRecord) {
+    // Most HoodieRecords are bound to have data + schema. Although, the same schema object is shared amongst
+    // all records in the JVM, hence to estimate the correct size,
+    // we calculate the size of the record by subtracting the size of the schema.
+    // NOTE : It may be the case that a HoodieRecord does not hold an instance of a schema object.
+    // So, for now, look for any object in the payload which might hold the schema, if such an object is found,
+    // return sizeOfRecord - sizeOfSchema else return the sizeOfRecord
+    // TODO : Find a generic way to figure out the true size of the record in any scenario
+    long sizeOfRecord = ObjectSizeCalculator.getObjectSize(hoodieRecord);
+    long sizeOfSchema = ObjectSizeCalculator.getObjectSize(schema);
+    log.info("SizeOfRecord => " + sizeOfRecord);
```

Comment:
single log statement.. also should this be info?

Reply:
Yeah, INFO seems fine, it's only logged once right now since the size estimate is done once.